### ISSUE TITLE
actions: It should be possible to pass deliver an event object as result in an action handler

### ISSUE THREAD

**Issue by alex3683** [LaxarJS/laxar_patterns#33]
Currently an action handler can only return whether its outcome is successful or not. This is the same for synchronous as for asynchronous handlers.
As an enhancement it should be possible to not only yield an outcome, but define a complete event object to be send along the `didTakeAction` event.

**Comment by alex3683** [comment]
## New Feature

In case of a successful outcome, it is also possible to return an object in the synchronous case or, in
case of an asynchronous handler, pass an object to the `done`-callback or the promise resolve function
respectively. This object will the be used as the blueprint for the event object of the `didTakeAction`
event.

Example:
```
actions.handlerFor( scope )
   .registerActions( [ 'open' ], function( event, meta ) {
      somethingSynchronous();
      return { response: 42 };
   } )
   .registerActions( [ 'save' ], function( event, meta, done ) {
      somethingAsynchronous()
         .then( function() {
            done( { response: 42 } );
         }, function() {
            done( actions.OUTCOME_ERROR );
         } );
   }, {  async: true } )
   .registerActions( [ 'save' ], function( event, meta, done ) {
      return q.when( { response: 42 } );
   }, { async: true  } );
```

**Comment by alex3683** [comment]
Implemented on `master` (`v0.20.0`).
